app/src/main/java/com/duckduckgo/app/survey/api/SurveyRepository.kt

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2023 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.survey.api
+
+import androidx.core.app.NotificationManagerCompat
+import com.duckduckgo.app.notification.NotificationRegistrar.NotificationId
+import com.duckduckgo.app.survey.db.SurveyDao
+import com.duckduckgo.app.survey.model.Survey
+import com.duckduckgo.browser.api.UserBrowserProperties
+import com.duckduckgo.di.scopes.AppScope
+import com.squareup.anvil.annotations.ContributesBinding
+import java.util.*
+import javax.inject.Inject
+
+interface SurveyRepository {
+    fun isUserEligibleForSurvey(survey: Survey): Boolean
+    fun remainingDaysForShowingSurvey(survey: Survey): Long
+    fun getScheduledSurvey(): Survey
+    fun clearSurveyNotification()
```

Comment:
This is confusing, we have a `clearSurveyNotification` but I can't see in this API where the notification is created

Reply:
I understand your concern. There were few options on how to approach this:

1. Inject `notificationManager` to SurveyViewModel directly so I can clear it from there

- It could work, but I believe it's better to have it in the Repository in case we want to perform different actions in the future, such as launching a pixel or interacting with other survey feature classes. This will ensure consistency if the function is called from different places in the future as well.

2. Add the function to the SurveyNotificationScheduler API and inject `notificationManager` there

- Because the class is a Scheduler instead a notification manager I thought adding the function here could be a bit confusing. However, it maybe makes more sense.

3. Inject `notificationManager` in the Repository like I did

- I went for this approach because as I explained in point 1, it's a way to unify the same behaviour for clearing survey notification from different places in the code. Also, it would be easier if we want to add a pixel or put survey in their own module in the future as a public API. However, it could also scale if I add the fun in the SurveyNotificationScheduler if you think it makes more sense @aitorvs . 

As a summery, I didn't add `clearNotification()` to the `NotificationScheduler` because it was a bit confusing for me as it is a Scheduler instead a Manager. But on the other hand, as you mentioned, it is also confusing to have the clearNotification function in the Repository and not having the create one. So I would agree on moving it to the NotificationScheduler if you think it's a better approach.

Reply:
No strong opinions, javadocs work as well. At least to understand what are we clearing and where it originates

Reply:
@nalcalag  check https://github.com/duckduckgo/Android/blob/develop/saved-sites/saved-sites-api/src/main/java/com/duckduckgo/savedsites/api/SavedSitesRepository.kt for api docs